packages/services/src/state_listener.rs

```diff
@@ -40,17 +42,24 @@ where
     async fn check_non_finalized_txs(&mut self, non_finalized_txs: Vec<L1Tx>) -> crate::Result<()> {
         let current_block_number: u64 = self.l1_adapter.get_block_number().await?.into();
 
+        let mut skip_nonces = HashSet::new();
+
+        let mut selective_change = vec![];
+        let mut noncewide_changes = vec![];
```

Comment:
Probably you guys are gonna do this soon, but just to make sure: can we get some comments here explaining the rationale for these two vecs and the strategy we're using here?